[quant] Add benchmakrs for embedding_bag coversion ops

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#43291 [quant] Add benchmakrs for embedding_bag coversion ops**

Summary:
Test Float2Fused and Fused2Float conversion operators for embedding_bag byte and 4-bit ops

Test Plan:

```
python -m pt.qembedding_pack_tes
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23231641](https://our.internmc.facebook.com/intern/diff/D23231641)

supriyar committed:
[quant] Add benchmakrs for embedding_bag coversion ops

Summary:
Test Float2Fused and Fused2Float conversion operators for embedding_bag byte and 4-bit ops

Test Plan:

```
python -m pt.qembedding_pack_tes
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Add benchmakrs for embedding_bag coversion ops"


Summary:
Test Float2Fused and Fused2Float conversion operators for embedding_bag byte and 4-bit ops

Test Plan:

```
python -m pt.qembedding_pack_tes
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant] Add benchmakrs for embedding_bag coversion ops"


Summary:
Test Float2Fused and Fused2Float conversion operators for embedding_bag byte and 4-bit ops

Test Plan:

```
python -m pt.qembedding_pack_tes
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

radkris-git commented on benchmarks/operator_benchmark/pt/qembedding_pack_test.py:
```diff
@@ -0,0 +1,52 @@
+
+embeddingbag_conversion_configs = op_bench.cross_product_configs(
+    num_embeddings=(100, 80, 120, 1000),
+    embedding_dim=(16, 4, 16, 64, 128),
```
Can we also add embedding_dims = 256, 512, 1024 and 2048?

radkris-git commented on benchmarks/operator_benchmark/pt/qembedding_pack_test.py:
```diff
@@ -0,0 +1,52 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
```
Thanks for sending this out.

Can you please answer these questions?
1. What does this benchmark test actually test? Does it just measure regression to the operator over time? 
2. How does the coverage looks like w.r.t. FBGeMM kernel?
3. Can we also add 2bit ops to the benchmark?

supriyar replied:
> What does this benchmark test actually test? Does it just measure regression to the operator over time?
Yes, this will track changes to the operator performance over time, it is tracked using AIBench and logged internally

> How does the coverage looks like w.r.t. FBGeMM kernel?
Once we make the operator use FBGEMM kernel the performance should improve and it will show up in the regression runs then.

> Can we also add 2bit ops to the benchmark?
Will do

supriyar committed:
Update on "[quant] Add benchmakrs for embedding_bag coversion ops"


Summary:
Test Float2Fused and Fused2Float conversion operators for embedding_bag byte and 4-bit ops

Test Plan:

```
python -m pt.qembedding_pack_tes
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

radkris-git commented:
LGTM.

mingzhe09088 commented:
Could you add this bench to https://github.com/pytorch/pytorch/blob/master/benchmarks/operator_benchmark/benchmark_all_quantized_test.py so it will be tested automatically?

supriyar committed:
Update on "[quant] Add benchmakrs for embedding_bag coversion ops"


Summary:
Test Float2Fused and Fused2Float conversion operators for embedding_bag byte and 4-bit ops

Test Plan:

```
python -m pt.qembedding_pack_tes
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
